### PR TITLE
[VisualTesting] Fix max length of dump name, add possible to create dump subfolders

### PR DESCRIPTION
**Pull request by MikhailKalyn**
Create from https://github.com/aquality-automation/aquality-selenium-core-dotnet/pull/95
Fix conversations

### CODE REVIEW

**Review by mialeska — CHANGES_REQUESTED** [review]

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/DumpManager.cs [hunk 1]
```diff
@@ -39,10 +44,10 @@ public virtual float Compare(string dumpName = null)
             {
                 throw new InvalidOperationException($"Dump directory [{directory.FullName}] does not exist.");
             }
-            var imageFiles = directory.GetFiles($"*{ImageFormat}");
+            var imageFiles = directory.GetFiles($"*.{ImageExtension.ToString().ToLower()}");
```
[the remark this answers is not in the document]
Comment (MikhailKalyn) [reply Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/DumpManager.cs:47]: There are actually no differences between the JPG and JPEG formats. The only difference is the number of characters used.
System.Drawing.Imaging.ImageFormat class doesn't contain ImageFormat member for Jpg, only Jpeg

same issue for .tif and .tiff formats

if we use the "save" method, then the dump will not contain different formats, only .jpeg
and if the dump is changed manually from the outside, then, in my opinion, this shouldn't affect the work of the framework and method 'compare'
it is the responsibility of the person preparing the dump manually

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/ImageExtensions.cs [hunk 2]
```diff
@@ -68,5 +69,35 @@ public static Image GetGrayScaleVersion(this Image original)
 
             return newBitmap;
         }
+
+        /// <summary>
+        /// Convert an image format from string representation to ImageFormat
+        /// </summary>
+        /// <param name="format">String representation of image format</param>
+        /// <returns>ImageFormat version of string representation</returns>
+        public static ImageFormat ConvertImageFormat(string format)
```
[the remark this answers is not in the document]
Comment (MikhailKalyn) [reply Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/ImageExtensions.cs]: fix

File: Aquality.Selenium.Core/tests/Aquality.Selenium.Core.Tests/Visualization/FormDumpTests.cs [hunk 3]
```diff
@@ -87,6 +97,85 @@ public void Should_BePossibleTo_SaveAndCompareWithDump_WithCustomName_WhenAllEle
...
+            foreach (var file in pathToDump.GetFiles())
+            {
+                Assert.AreEqual(imageExtension, file.Extension, "Image extension not exual to expexted");
+                Assert.That(file.Name.Contains(imageExtension), "Image name doesn't contain expected extension");
+            }
+        }
+
+        [TestCase(".abc")]
+        [TestCase("")]
+        public void Should_BePossibleTo_SaveFormDump_WithInvalidExtension(string imageExtension)
```
[the remark this answers is not in the document]
Comment (MikhailKalyn) [reply Aquality.Selenium.Core/tests/Aquality.Selenium.Core.Tests/Visualization/FormDumpTests.cs]: fixed

File: Aquality.Selenium.Core/tests/Aquality.Selenium.Core.Tests/Visualization/FormDumpTests.cs [hunk 4]
```diff
@@ -87,6 +97,85 @@ public void Should_BePossibleTo_SaveAndCompareWithDump_WithCustomName_WhenAllEle
...
+
+        private DirectoryInfo CleanUpAndGetPathToDump(string dumpName)
+        {
+            var pathToDump = new DirectoryInfo(Path.Combine(PathToDumps, dumpName));
+            if (pathToDump.Exists)
+            {
+                pathToDump.Delete(true);
+                pathToDump.Refresh();
+            }
+            Assert.AreEqual(0, pathToDump.Exists? pathToDump.GetFiles().Length : 0, "Dump directory should not contain any files before saving");
```
[the remark this answers is not in the document]
Comment (MikhailKalyn) [reply Aquality.Selenium.Core/tests/Aquality.Selenium.Core.Tests/Visualization/FormDumpTests.cs]: fixed

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/DumpManager.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mialeska) [reply Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/DumpManager.cs]: It's all about consistancy. If I have ".jpg" extension in settings.json, I expect that files with this extension could be used for comparison. If I tell to save in ".jpg" extension but framework saves in ".jpeg" anyways, it looks like a bug for me. ".jpg" extension is used way more often than ".jpeg" in the real world

[the remark this answers is not in the document]
Comment (MikhailKalyn) [reply Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/DumpManager.cs]: fixed

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/ImageFormat.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+        private ImageFormat(string format) { this.format = format; }
+        private ImageFormat(DrawingImageFormat format) { this.format = format.ToString(); }
+
+        /// <summary>
+        /// Convert an image format from string representation to ImageFormat
+        /// </summary>
+        /// <param name="format">String representation of image format</param>
+        /// <returns>ImageFormat version of string representation</returns>
+        public static ImageFormat Convert(string format)
```
Comment (mialeska) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/ImageFormat.cs:31]: looks like "Parse"

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/DumpManager.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MikhailKalyn) [reply Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/DumpManager.cs]: fixed

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/ImageFormat.cs (hunk 5 above)
Comment (MikhailKalyn) [reply Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/ImageFormat.cs]: fixed

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/ImageFormat.cs [hunk 6]
```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using DrawingImageFormat = System.Drawing.Imaging.ImageFormat;
```
[the remark this answers is not in the document]
Comment (MikhailKalyn) [reply Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/ImageFormat.cs:2]: fixed

**Review by mialeska — APPROVED** [review]

**Pull request merged**